src/clj_foundation/errors.clj

```diff
@@ -109,3 +106,65 @@ attempts, the last caught exception is re-thrown."
   The remaining parameters (if any) are that function's arguments."
   [tries pause-millis]
   (fn [f & args] (apply retry tries pause-millis f args)))
+
+
+(s/defn replace-if :- s/Any
```

Comment:
If it's only needed within icarrp, then it should have stayed there.  Once it was extracted to it's own library, it became something that can affect multiple aspects of BEETL and potential future projects.  So where it lives *is* relevant and its design becomes harder to change with every dependent it gains (which are tracked separately *because* of where it lives).  I will accept a TD ticket as resolution to this.